src/libraries/AccountAccessParser.sol

```diff
@@ -207,27 +207,28 @@ library AccountAccessParser {
         // The order of 'uniqueAccounts' informs the order that the account state diffs get processed.
         address[] memory uniqueAccounts = getUniqueWrites(_accountAccesses, _sort);
         uint256 totalDiffCount = 0;
+        StateDiff[][] memory accountDiffs = new StateDiff[][](uniqueAccounts.length);
```

Comment:
Sgtm!

Reply:
ah, before we had:
```solidity
for (uint256 i = 0; i < uniqueAccounts.length; i++) {
            StateDiff[] memory accountDiffs = getStateDiffFor(_accountAccesses, uniqueAccounts[i], _sort);
```
so `getStateDiff` was called `uniqueAccounts.length` times. 

This means, we want an array of `StateDiff[]` that's `uniqueAccounts.length` in size.